sim-lib/src/lib.rs

```diff
@@ -337,6 +342,19 @@ pub struct Simulation {
     no_results: bool,
 }
 
+/// PayGenDisplay is a marker trait to allow creation of trait objects for printable payment generators.
+pub trait PayGenDisplay: PaymentGenerator + Display {}
+
+/// ExecutorKit contains the components required to spin up an activity configured by the user, to be used to
+/// spin up the appropriate producers and consumers for the activity.
+struct ExecutorKit {
+    source_info: NodeInfo,
+    /// We use an arc mutex here because some implementations of the trait will be very expensive to clone.
+    /// See [NetworkGraphView] for details.
+    network_generator: Arc<Mutex<dyn NetworkGenerator + Send>>,
```

Comment:
I think this can also be a `Box`.

We are not accessing this from more than one thread at the same time, and we are already requiring it to be `Send`